qa/tasks/vstart_runner.py

```diff
@@ -394,6 +395,26 @@ def sh(self, command, log_limit=1024, cwd=None, env=None):
         return misc.sh(command=command, log_limit=log_limit, cwd=cwd,
                         env=env)
 
+    @property
```

Comment:
could you note in the code as comment that this hunk is copied from `teuthology/teuthology/orchestra/remote.py` . or we could use a kinda hackish approach to avoid copy-n-pasting, like

```py
from teuthology.orchestra.opsys import Remote

# ...

class LocalDaemon(object):
  # reuse these methods in my "Remote" cousin
  os = Remote.os
  arch = Remote.arch
```

Reply:
Yeah, sure will do it.
Thanks @tchaikov

Reply:
The os/arch are property, not methods, the reuse seem not work.

Reply:
they are methods.

Reply:
this is a minimal demo:

```py
class Remote:
    @property
    def os(self):
        try:
            return self._os
        except AttributeError:
            self._os = 'hello'
            return self._os

class Local:
    os = Remote.os


l = Local()
print(l.os)
```

Reply:
Ｉ had tried this in the morning, and I got the following errors:

```
    @property
    def os(self):
        self._os = Remote.os
        return self._os

    @property                                                                                                                                     
    def arch(self):
        self._arch = Remote.arch
        return self._arch

```
```

2020-03-06 09:18:30,068.068 INFO:__main__:
2020-03-06 09:18:30,068.068 INFO:__main__:======================================================================
2020-03-06 09:18:30,069.069 INFO:__main__:ERROR: test_filelock (tasks.cephfs.test_client_recovery.TestClientRecovery)
2020-03-06 09:18:30,069.069 INFO:__main__:----------------------------------------------------------------------
2020-03-06 09:18:30,069.069 INFO:__main__:Traceback (most recent call last):
2020-03-06 09:18:30,070.070 INFO:__main__:  File "/data/ceph/qa/tasks/cephfs/test_client_recovery.py", line 397, in test_filelock
2020-03-06 09:18:30,070.070 INFO:__main__:    flockable = self._is_flockable()
2020-03-06 09:18:30,070.070 INFO:__main__:  File "/data/ceph/qa/tasks/cephfs/test_client_recovery.py", line 371, in _is_flockable
2020-03-06 09:18:30,070.070 INFO:__main__:    a_version_str = get_package_version(self.mount_a.client_remote, "fuse")
2020-03-06 09:18:30,071.071 INFO:__main__:  File "/data/teuthology/teuthology/packaging.py", line 351, in get_package_version
2020-03-06 09:18:30,071.071 INFO:__main__:    if remote.os.package_type == "deb":
2020-03-06 09:18:30,071.071 INFO:__main__:AttributeError: 'property' object has no attribute 'package_type'
2020-03-06 09:18:30,071.071 INFO:__main__:

```
Is there anything that I missed ?

Reply:
i think both of them are wrong.

what i suggest is
```py
class Local:
    os = Remote.os
```

Reply:
Okay, it works now. fixed it 
Thanks @tchaikov